minetest/minetest
# Move unsupported language list into a separate file

**Goal of the PR**
This PR tries to prevent compilation of unsupported languages on Android build (fixes #13808).
~This PR tries to add better handling for unsupported languages (fixes #13808).~ _(see replies below)_

**How does the PR work?**
- Move unsupported language list into a separate file. Both build scripts now read from that file.
- ~Add unsupported language list in the `build.gradle`~ _(see replies below)_
- ~Add more checks when initialising language for Gettext~ _(see replies below)_

**Does it resolve any reported issue?**
Yes, #13808.

**Does this relate to a goal in [the roadmap](https://github.com/minetest/minetest/blob/master/doc/direction.md)?**
The reported issue is in the 5.8.0 milestone.

## To do

This PR is Ready for Review.

## How to test

1. Set system language to be any language in the unsupported language list.
2. Run Minetest.
3. Make sure that the shown language is in English.
4. Retry step 1 to 3, but using (a) `language` setting in `minetest.conf` and (b) environment variables (`LANG` or `LANGUAGE`).

## Notes

Implementation (code) suggestions are welcome. I just tried to piece together some code that might/might not work for all operating system. I have tested this PR in Windows 10 and Android 13.

Some suggest to only change the build steps so that unsupported languages are never selected (because they do not exist). If that is the better case, any changes to `src/gettext.(cpp|h)` can be canceled.
